src/storage-client/src/controller.rs

```diff
@@ -2051,9 +2051,25 @@ where
             Some(StorageResponse::FrontierUppers(updates)) => {
                 self.update_write_frontiers(&updates);
             }
-            Some(StorageResponse::DroppedIds(_ids)) => {
-                // TODO(petrosagg): It looks like the storage controller never cleans up GlobalIds
-                // from its state. It should probably be done as a reaction to this response.
+            Some(StorageResponse::DroppedIds(ids)) => {
+                let shards_to_finalize: Vec<_> = ids
```

Comment:
Double-checking a problem we've had in the past: do we get `DroppedIds` messages for all the subsources as well, or do _we_ have to make sure to register them here?

~~Also, I think that TODO is still valid? We still don't delete them from our collection state, do we?~~ Ah no, `finalize_shards`, does that right?

Reply:
1. Yes, we do get the subsource IDs here, as well.
2. This TODO should not have been dropped here because the do not remove the `GlobalId` from `self.state.collections`; all of this code got taken from a branch that _does_ do this work, which was why it got removed. Sorry for the confusion here.